plugin/storage/es/spanstore/dbmodel/to_domain.go

```diff
@@ -12,27 +12,35 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package json
+package dbmodel
 
 import (
 	"encoding/hex"
 	"fmt"
 	"strconv"
+	"strings"
 
 	"github.com/pkg/errors"
 
 	"github.com/jaegertracing/jaeger/model"
-	"github.com/jaegertracing/jaeger/model/json"
 )
 
-// SpanToDomain converts json.Span with embedded Process into model.Span format.
-func SpanToDomain(span *json.Span) (*model.Span, error) {
-	return toDomain{}.spanToDomain(span)
+// NewToDomain creates ToDomain
+func NewToDomain(deDotChar string) ToDomain {
+	return toDomain{deDotChar: deDotChar}
 }
 
-type toDomain struct{}
+// ToDomain is used to convert Span to model.Span
+type ToDomain interface {
```

Comment:
Same question as before: is this specific to ES? Couldn't this be made at a higher level, to be used by other storage backends? Otherwise, what's the point of changing from a struct to an interface?

Reply:
The biggest advantage is keeping the struct private and exposing only the interface with builder function. It makes it clear how it is constructed.